Node/Generator/generators/app/index.js

```diff
@@ -24,7 +23,7 @@ module.exports = class extends Generator {
     const botName = this.props.botName;
     const extension = this.props.language === 'JavaScript' ? 'js' : 'ts';
     const launchSteps = extension === 'js' ? `node app.js` : `tsc\nnode app.js`;
-    const defaultDialog = this.props.dialog.split(' ')[0].toLowerCase();
+    const defaultDialog = this.props.dialog ? this.props.dialog.split(' ')[0].toLowerCase() : this.props.dialog;
```

Comment:
I don't understand this change. Can you add a comment explaining why you're making it?

Reply:
This code has been updated and comments added in latest version of PR.